File: Leetcode/PascalsTriangleII.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<int> a, b;
        a.push_back(1);
        for(int i = 0; i < rowIndex - 1; ++i){
            b.clear();
            b.push_back(1);
            for(int j = 1; j < a.size(); ++j){
                b.push_back(a[j - 1] + a[j]);
            }
            if(i % 2 == 0) b.push_back(2 * a[(int)a.size() - 1]);
            swap(a, b);
        }
        b = a;
        int st = (rowIndex % 2 == 1 ? a.size() - 1 : a.size() - 2);
        for(; st >= 0; --st) a.push_back(b[st]);
        return a;
    }
};

File: Leetcode/PascalsTriangleII.rs
// https://leetcode.com/problems/pascals-triangle-ii

struct Solution {}

impl Solution {
    pub fn get_row(row_index: i32) -> Vec<i32> {
        let mut cur = vec![1];

        (0..row_index).for_each(|_| {
            cur = cur.iter().enumerate().fold(vec![1], |mut next, (j, x)| {
                next.push(x + cur.get(j + 1).unwrap_or(&0));
                next
            });
        });

        cur
    }
}
